Switch to next server on Ctrl+Tab

kanishk98 opened the pull request:
<!--
Remove the fields that are not appropriate
Please include:
-->

**What's this PR do?**

Adds a new option to the Window menu that allows users to change to the
next organization cyclically. Activated by Ctrl/Cmd + Tab.

**Any background context you want to provide?**

Please have a look at #659. 

**Screenshots?**

![ezgif com-crop](https://user-images.githubusercontent.com/24617297/53291604-dad0e700-37db-11e9-9f6b-0b9c04f2eb68.gif)

**You have tested this PR on:**
  - [x] Windows
  - [x] Linux/Ubuntu
  - [x] macOS

kanishk98 committed:
Switch to next server on Ctrl+Tab

Adds a new option to the Window menu that allows users to change to the
next organization cyclically. Activated by Ctrl/Cmd + Tab.

akashnimare commented:
This looks good. Will merge it after the new release.

abhigyank commented:
See if you can also include Ctrl+Shift+Tab to go to previous server.

kanishk98 commented:
@abhigyank sure, shouldn’t be a problem. I’ve got an exam this morning, will finish this up once I get back. :)

Also, just a suggestion, we use Ctrl + Shift + T to restore a tab we accidentally closed in the browser. Would it be a good idea to include this as a shortcut too, in case someone accidentally deletes an organisation (now that we’re adding more options to the context menu in the sidebar, that seems possible)

abhigyank commented:
I don't suppose deleting an organization accidently is much of a common use case and could be confusing for users.

kanishk98 commented:
> I don't suppose deleting an organization accidently is much of a common use case and could be confusing for users.

Fair enough. We can always work on it later depending on feedback.

kanishk98 committed:
Switch to previous server on Ctrl+Shift+Tab

Adds menu item and keyboard shortcut for switching to the server above
the current one in the sidebar. Does not allow user to go cycle beyond
the first server.

kanishk98 committed:
Add shortcut listings to shortcuts section

Ctrl/Cmd + Tab and Ctrl/Cmd + Shift + Tab added to the list of keyboard
shortcuts in the preferences section of the app.

akashnimare commented:
Okay, this doesn't work on macOS. Maybe because the `CMD + Tab` shortcut is globally registered to switching between apps. @priyank-p any idea?

kanishk98 commented:
> Okay, this doesn't work on macOS. Maybe because the `CMD + Tab` shortcut is globally registered to switching between apps. 

Right, makes sense. I checked [Chrome keyboard shortcuts](https://support.google.com/chrome/answer/157179?hl=en) and they seem to use ⌘ + Option + Right arrow for switching to the next tab. Would it be a good idea to do the same here?

akashnimare commented:
> Right, makes sense. I checked Chrome keyboard shortcuts and they seem to use ⌘ + Option + Right arrow for switching to the next tab. Would it be a good idea to do the same here?

CTRL + Tab also works for switching the tabs. Let's use the same in our app also. i.e. CTRL + Tab on all the OSes.

akashnimare commented on app/renderer/js/pages/preference/shortcuts-section.js:
```diff
@@ -130,6 +130,14 @@ class ShortcutsSection extends BaseSection {
                     <td>Toggle DevTools for Active Tab</td>
                   </tr>
+                  <tr>
+                    <td><kbd>${userOSKey}</kbd> + <kbd>Tab</kbd></td>
```
This should be CTRL.

akashnimare commented on app/renderer/js/pages/preference/shortcuts-section.js:
```diff
@@ -130,6 +130,14 @@ class ShortcutsSection extends BaseSection {
+                    <td>Switch to next organization</td>
+                  </tr>
+                  <tr>
+                    <td><kbd>${userOSKey}</kbd> + <kbd>Shift</kbd> + <kbd>Tab</kbd></td>
```
Same.

kanishk98 replied:
I've pushed the changes needed. I don't have access to a Mac right now, so I can't check for myself if the section looks as expected. Please let me know if it doesn't.

kanishk98 committed:
Change Cmd to Ctrl for macOS

Since Cmd + Tab is registered in macOS as the shortcut to switch apps,
shortcut needs to be changed to Ctrl + Tab for all OSes.

Replace Cmd with Ctrl in Shortcuts Section

akashnimare commented:
Merged. Thanks, @kanishk98.

abhigyank commented:
I think we forgot to change userOSKey to Ctrl in settings.

akashnimare commented:
We did change it for the macOS, for Windows/Linux userOSkey is fine.
